Scripts/Util/TextureManager.cs

```diff
@@ -12,27 +15,30 @@ public static Texture PackTextures(Texture[] textures, out Rect2[] uvs)
         uvs = new Rect2[textures.Length];
 
         Image atlasImage = new Image();
-        atlasImage.Create(images[0].GetWidth() * images.Length, images[0].GetHeight(), false, Image.Format.Rgb8);
+        atlasImage.Create(images[0].GetWidth() * images.Length + (MARGIN * images.Length + 1), images[0].GetHeight(), false, Image.Format.Rgb8);
         atlasImage.Lock();
+        int xPos = MARGIN / 2;
+        // We will repeat the texture into the margins, so if there are any floating point inaccuracies in the UVs, they point into valid pixels of the atlas
         for(int i = 0; i < images.Length; i++)
         {
-            uvs[i] = new Rect2((float)i / images.Length, 0, 1f / images.Length, 1);
+            uvs[i] = new Rect2((float)xPos / atlasImage.GetWidth(), 0, (float)images[0].GetWidth() / atlasImage.GetWidth(), 1);
 
             images[i].Lock();
-            for(int x = 0; x < images[i].GetWidth(); x++)
+            for(int x = -MARGIN; x < images[i].GetWidth() + MARGIN; x++)
             {
                 for(int y = 0; y < images[i].GetHeight(); y++)
                 {
-                    atlasImage.SetPixel((images[0].GetWidth() * i) + x, y, images[i].GetPixel(x, y));
+                    atlasImage.SetPixel(xPos + x, y, images[i].GetPixel((x + images[i].GetWidth()) % images[i].GetWidth(), y));
```

Comment:
why SetPixel and iteration instead of BlitRect ?

Reply:
Didn't realise that was a thing - will change